---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# MeTPeak

## Update

Update the original package to compile with the current Bioconductor packages. (2024/09/16)

> Modify following script:
>
> `R/get.check.points.reads.count.R`
>
> `R/read.gtf.R`

Below is the example from MeTPeak

```{r,eval=FALSE}
devtools::install_github("thereallda/MeTPeak")
```

```{r}
library(MeTPeak)

# in the real case, change the gtf to what you need
gtf <- system.file('extdata','example.gtf',package='MeTPeak')
# tmp <- rtracklayer::import(gtf)

ip1 <- system.file('extdata','IP1.bam',package='MeTPeak')
ip2 <- system.file('extdata','IP2.bam',package='MeTPeak')
ip3 <- system.file('extdata','IP3.bam',package='MeTPeak')
input1 <- system.file('extdata','Input1.bam',package='MeTPeak')
input2 <- system.file('extdata','Input2.bam',package='MeTPeak')
input3 <- system.file('extdata','Input3.bam',package='MeTPeak')

IP_BAM <- c(ip1,ip2,ip3)
INPUT_BAM <- c(input1,input2,input3)

metpeak(GENE_ANNO_GTF=gtf,
        IP_BAM = IP_BAM,INPUT_BAM = INPUT_BAM,
        EXPERIMENT_NAME="example",
        OUTPUT_DIR = "./")
```

After running `metpeak`, results will be directed to the folder (named in `OUTPUT_DIR/EXPERIMENT_NAME`). In the current case, results are stored at "./example" directory.

`MeTPeak` peak calling results stored at "peak.bed" and "peak.xls"

```{r}
# load output
load("example/metpeak.Rdata")
names(tmp_rs) # tempoary results
# peak results
metpeak_df <- read.table("example/peak.bed", sep="\t")
# cols: chr	chromStart	chromEnd	name	score	strand	thickStart	thickEnd	itemRgb	blockCount	blockSizes	blockStarts	lg.p	lg.fdr	fold_enrchment
head(metpeak_df)
```
